examples/basecoin/types/account.go

```diff
@@ -27,6 +27,9 @@ func GetParseAccount(cdc *wire.Codec) sdk.ParseAccount {
 	return func(accBytes []byte) (res sdk.Account, err error) {
 		acct := new(AppAccount)
 		err = cdc.UnmarshalBinary(accBytes, acct)
+		if err != nil {
+			panic(err)
+		}
```

Comment:
why the panic in here when it returns an error ?

Reply:
I was discussing with @jaekwon and it was his opinion which I'm coming around too is that an error caused by unmarshal bytes, where the software was the creator of those bytes is a panic condition - because it means there something wrong/unexpected it the software which is causing it to write improper bytes